File: day03/triangles.pl
#!/usr/bin/env perl
use strict;
use warnings;
use 5.24.0;

my $num_good = 0;
while (<>) {
    chomp;
    s/^\s+//;
    my @s = split /\s+/;
    my $bad = 0;
    for my $i (0..1) {
        for my $j ($i+1..2) {
            if ($s[$i] + $s[$j] <= $s[3-$i-$j]) {
                $bad = 1;
                last;
            }
        }
    }
    $num_good++ unless $bad;
}

say $num_good;

File: day03/triangles.py
#!/usr/bin/env python
from sys import argv
from numpy import *

num_good = 0

script, filename = argv
with open(filename) as f :
    for line in f :
        line = line.rstrip("\n")
        s = map(int, line.split())
        bad = False
        for i in range(0, 2) :
            for j in range(i+1, 3) :
                if s[i] + s[j] <= s[3-i-j] :
                    bad = True
                    break
        if not bad :
            num_good = num_good + 1

print num_good

